components/SelectComponent.tsx

```diff
@@ -0,0 +1,99 @@
+import { fhirJson } from "@fhir-typescript/r4-core";
+import { useState, useEffect, Dispatch, SetStateAction } from "react";
+import { Autocomplete } from "@mantine/core";
+
+/**
+ * @param props include the string resourceType
+ * @returns a component with a loading component, server error, or autocomplete select component populated with resource IDs
+ */
+export default function SelectComponent(props: {
+  resourceType: string;
+  setPractitionerValue: Dispatch<SetStateAction<string>>;
+  practitionerValue: string;
+}) {
+  const resourceType = props.resourceType;
+  const [pageBody, setPageBody] = useState<fhirJson.Bundle>();
+  const [fetchingError, setFetchingError] = useState(false);
+  const [loadingRequest, setLoadingRequest] = useState(false);
+
+  useEffect(() => {
+    if (resourceType) {
+      fetch(`${process.env.NEXT_PUBLIC_DEQM_SERVER}/${resourceType}`)
+        .then((data) => {
+          return data.json() as Promise<fhirJson.Bundle>;
+        })
+        .then((resourcePageBody) => {
+          setPageBody(resourcePageBody);
+          setFetchingError(false);
+          setLoadingRequest(false);
+        })
+        .catch((error) => {
+          console.log(error.message);
+          setFetchingError(true);
+          setLoadingRequest(false);
+        });
+    }
+  }, [resourceType]);
+
+  return loadingRequest ? (
+    <div>Loading content...</div>
+  ) : !fetchingError && pageBody ? (
```

Comment:
never been a huge fan of chaining ternary operators from a readability standpoint. You could just break out these situations into if statements, each with their own return statement inside. Thoughts?

Reply:
Similar response here as with the .then comment--happy to make the change, but a lot of other docs in our code use the same format. If I change this one, should I change those as well?

Reply:
Lol true. very much a preference thing for me, so I'm happy to just leave it as is